microsoft-edge/extensions-chromium/developer-guide/sidebar.md

```diff
@@ -0,0 +1,625 @@
+
+To allow users to open the sidebar by clicking the action toolbar icon, use [sidePanel.setPanelBehavior()](https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-setPanelBehavior).  First, declare the `"action"` key in the manifest:
+
+`manifest.json`:
+
+```json
+{
+  "name": "My sidebar extension",
+  ...
+   "action": {
+    "default_title": "Click to open sidebar"
+  },
+  ...
+}
+```
+
+Then, add the following code to the code listing that's in [Enable a sidebar for a specific site only](#enable-a-sidebar-for-a-specific-site-only) above:
+
+`service-worker.js`:
+
+```js
+const BING_ORIGIN = 'https://www.bing.com';
+
+// Allow users to open the sidebar by clicking on the action toolbar icon
+chrome.sidePanel
+  .setPanelBehavior({ openPanelOnActionClick: true })
+  .catch((error) => console.error(error));
+...
+```
+
+
+<!-- ------------------------------ -->
+#### Switch to a different sidebar
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#multi-panels -->
+
+An extension can use [sidepanel.getOptions()](https://developer.chrome.com/docs/extensions/reference/sidePanel/#method-getOptions) to retrieve the current sidebar, and then enable a different sidebar for a specific tab.
+
+This example sets a sidebar containing a welcome message on [runtime.onInstalled()](https://developer.chrome.com/docs/extensions/reference/runtime/#event-onInstalled).  When the user navigates to a different tab, the sidebar is replaced with the browser-level sidebar.
+
+`service-worker.js`:
+
+```js
+const welcomePage = 'sidebar/welcome-sp.html';
+const mainPage = 'sidebar/main-sp.html';
+
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.sidePanel.setOptions({ path: welcomePage });
+});
+
+chrome.tabs.onActivated.addListener(async ({ tabId }) => {
+  const { path } = await chrome.sidePanel.getOptions({ tabId });
+  if (path === welcomePage) {
+    chrome.sidePanel.setOptions({ path: mainPage });
+  }
+});
+```
+
+
+<!-- ====================================================================== -->
+## The sidebar user experience (UX)
+
+<!-- compare https://developer.chrome.com/docs/extensions/reference/sidePanel/#user-experience -->
+
+Extensions in the Microsoft Edge sidebar have these user experience (UX) features.
+
+
+<!-- ------------------------------ -->
+#### How extensions that have sidebar support are indicated
+
+* The sidebar icon is displayed next to the extension name in the hub in the toolbar:
+
+![Sidebar dialog](./sidebar-images/sidebar-dialog.png)
+
+
+###### Using the action toolbar icon
+
+Users can open the sidebar by clicking the extension shortcut in the toolbar, if it's enabled.
+
+See [Enable the extension's shortcut icon to open the sidebar](#enable-the-extensions-shortcut-icon-to-open-the-sidebar), above.
+
+
+<!-- todo: un-comment & revise, after implemented:
+###### Using a keyboard shortcut
```

Comment:
This is implemented and can be uncommented

Reply:
done locally